nixpkgs/nixpkgs.bzl

```diff
@@ -185,3 +186,63 @@ stderr:
   code=exec_result.return_code,
   stdout=exec_result.stdout,
   stderr=exec_result.stderr))
+
+def _nixpkgs_cc_autoconf_impl(repository_ctx):
+  # Calling repository_ctx.path() on anything but a regular file
+  # fails. So the roundabout way to do the same thing is to find
+  # a regular file we know is in the workspace (i.e. the WORKSPACE
+  # file itself) and then use dirname to get the path of the workspace
+  # root.
+  workspace_file_path = repository_ctx.path(
+    Label("@nixpkgs_cc_toolchain//:WORKSPACE")
+  )
+  workspace_root = repository_ctx.execute(
+    ["dirname", workspace_file_path]
+  ).stdout.rstrip()
+
+  # Make a list of all available tools in the Nix derivation. Override
+  # the Bazel autoconfiguration with the tools we found.
+  res = repository_ctx.execute(["ls", workspace_root + "/bin"])
+  if res.return_code == 0:
+    overriden_tools = {
+      tool: repository_ctx.path(Label("@nixpkgs_cc_toolchain//:bin/" + tool))
+      for tool in res.stdout.splitlines()
+    }
+    cc_autoconf_impl(repository_ctx, overriden_tools = overriden_tools)
```

Comment:
The rule uses `repository_ctx.which` in `unix_cc_configure.bzl` to resolve paths passed through `overridden_tools`, and it’s not clear from the documentation what that means for the absolute paths we pass it: https://github.com/bazelbuild/bazel/issues/6660 (I’m too lazy to look into the Java source right now, but I assume it returns absolute paths unmodified).